node-core/components/signer.go

```diff
@@ -58,6 +61,17 @@ func ProvideBlsSigner(in BlsSignerInput) (crypto.BLSSigner, error) {
 		if !filepath.IsAbs(privValStateFile) {
 			privValStateFile = filepath.Join(homeDir, privValStateFile)
 		}
+
+		// Check key file existence here as the error in NewBLSSigner is vague.
+		if _, err := os.Stat(privValKeyFile); os.IsNotExist(err) {
+			return nil, fmt.Errorf("key file does not exist at path: %s", privValKeyFile)
+		}
+
+		// Check state file existence as the error in NewBLSSigner is vague.
+		if _, err := os.Stat(privValStateFile); os.IsNotExist(err) {
```

Comment:
It isnt absolutely necessary to have the state file. See the CLI commands which use this ProvideSigner function to determine the BLS signer. They should all be able to run without state

Reply:
Basically all uses of ProvideSigner should allow not having state
- when node starts up
- CLI deposit commands
- etc

Reply:
EDIT: After investigating, the `LoadFilePV` that we use currently does indeed require having the state file so this check is safe to keep as is.